starter-code/routes/index.js

```diff
@@ -1,9 +1,79 @@
 const express = require('express');
 const router  = express.Router();
+const multer  = require('multer');
+// Route to upload from project base path
+const upload = multer({ dest: 'uploads/' }) 
+const User = require('../models/user');
+const bcrypt = require('bcrypt');
+const bcryptSalt = 10;
+
+const passport = require("passport");
+const ensureLogin = require("connect-ensure-login");
+const Picture = require('../models/picture');
+ 
 
 /* GET home page. */
-router.get('/', (req, res, next) => {
-  res.render('index', { title: 'Express - Generated with IronGenerator' });
+router
+.get('/', (req, res, next) => {
+		res.render('index');
+});
+
+router
+.get('/signup', (req, res, next) => {
+		res.render('authentication/signup', { "message": req.flash("error") });
+});
+
+
+router
+.post('/signup', upload.single('photo'), function(req, res, next){
+
+	const photo = new Picture({
+		path: `/uploads/${req.file.filename}`,
+		originalName: req.file.originalname
+	});
+
+	const user = new User({
+		username: req.body.username,
+		email: req.body.email,
+		password: req.body.password,
+		photo: photo,
+	});
+
+	user.save((err) => {
+		photo.save((err) => {
+			res.render("authentication/profile", {user});
```

Comment:
You are rendering a new view but not rerouting the user to a new url. If you notice, the route remains /signup. It makes more sense to call res.redirect to a new url, and then handle the get request to that url seperately with a view render.